### ISSUE TITLE
Layout Serialization is causing problems due to regional formatting

### ISSUE THREAD

**Issue by Karmoq** [npolyak/NP.Avalonia.UniDock#6]
Hey Nick,

I'm having an issue with the serialization of layouts, and I believe it is related to regional formatting.
My colleague has czech regional formatting, which uses commas (",") for separating decimal numbers, instead of the usual decimal points (".")

After looking at the serialization, it seems like that when saving the layouts, it uses the GridLength.ToString() method to convert it to strings, which is not culture-invariant. See `DockGroupParents.ToGroupParams(this IDockGroup dg)`, line 149.

But when loading the layouts, it uses the `GridLength.Parse` method, which, when looking at the Avalonia Source code, correctly uses Invariant Culture for parsing.

In some cases this results in commas being ignored when loading a saved layout:

Before saving
![image](https://user-images.githubusercontent.com/27163398/168266577-5dedbaa0-6db9-47f2-b48a-a64fe1db961b.png)

After saving
![image](https://user-images.githubusercontent.com/27163398/168273561-1cdbde4f-c510-4f5a-816c-00f260d47540.png)

I believe this can be fixed by changing the non-culture-invariant call to `GridLength.ToString()` to something like `GridLength.Value.ToString(CultureInfo.Invariant)` (of course considering the star etc)

I hope it is clear what I'm explaining :)

Thanks,
Karmoq

**Comment by npolyak** [comment]
Sorry Karmoq, it took me some time to reply to you. 

In the USA, commas are not used for decimals - only for separation of thousands, Please use period - '.' for decimals:)

**Comment by Karmoq** [comment]
Hey Nick,
I think you didn't fully understand the issue, it seems that in with the way how Unidock serializes the layout, it uses culture dependent .ToString() for saving (which uses the machines regional formatting setting) while it uses Invariant parsing for loading.
This means that when the users machine is set to a different regional formatting, Unidock itself generates save files with different formatting than what it expects when loading.
Basically, users with different regional formatting on their machines can't use the layout serialization as it is right now.

**Comment by npolyak** [comment]
Yes, you are correct. Actually it is a problem with Avalonia's GridLength - they provide local ToString, but locale-invariant parse, but I can fix it and I will

**Comment by npolyak** [comment]
please, download the next version of UniDock - https://www.nuget.org/packages/NP.Avalonia.UniDock/1.0.37

It should be working.
